src/frontend.md

```diff
@@ -5,31 +5,54 @@
 ```
 src/
   components/ (can be dumb and smart)
-    bankAccount/
+    [All components should be placed in some entity folder or shared folder]
+    bankAccount/ (entity, module, .etc)
       BankAccountTable/
         [no nested directories]
+        [avoid index.ts]
```

Comment:
having `index.ts` doesn't protect us from breaking modularity, but requires developers' time and bundle size

Reply:
1) I agree with you that it does not protect modularity, but for this there are our principles and review code
2) to speed up the process of writing components: there are a lot of plugins for different IDEs
3) unfortunately we have much more critical situations with sizes, and the index.js is a drop in the ocean.

PS: I want this pattern to be respected at the component level - this is important in the long run
![image](https://user-images.githubusercontent.com/11217819/217632423-5a5df334-9ec0-460c-b879-10aa1ac1809f.png)

Reply:
so you mean if we're going to write `import { Whatever } from '../Whatever';` instead of `import { Whatever } from './Whatever';` allows us to have loose coulpling and hight cohesion? :)

Reply:
@viczhuravlev want to say, that if you don't have index file in your folder, you can't understand which elements of your folder/module are internal and which external
index file it's a good place where you can define all external elements of your folder, and can be entry point for your connection this module to others
As for me such example better
```
import { Component1, Component2, Type1, helper1 } from '../modules/module_1'
import { Component1, Component2, Type1, helper1 } from '../modules/module_2'
```
than it
```
import { Component1 } from '../modules/module_1/components/Component1'
import { Component2 } from '../modules/module_1/components/Component2'
import { Type1 } from '../modules/module_1/components/types'
import { helper1 } from '../modules/module_1/components/helpers/helper1'
import { Component1 } from '../modules/module_2/components/Component1'
import { Component2 } from '../modules/module_2/components/Component2'
import { Type1 } from '../modules/module_2/components/types'
import { helper1 } from '../modules/module_2/components/helpers/helper1'
```

Reply:
@Lecsar i guess if `Type1` and `helper1` are reused, they should be extracted out of component folder.

Exposing only external things is good, but we should focus on development speed. Usually I don't write imports manually, my IDE writes them for me, and IDE can write `import { Component1 } from '../modules/module_1/components/Component1'` instead of `import { Component1 } from '../modules/module_1'`. And I don't want to spend time on fixing it and supporting `index.ts` manually, it's just additional cost for Osome.

Reply:
@Rendalf for me it's history about internal/external things of module. If we have index file i clearly understand in one place what I export and reuse, and which interfaces I can't change because it's external/public file